executor/partition_table_test.go

```diff
@@ -756,6 +756,101 @@ func (s *partitionTableSuite) TestDirectReadingWithAgg(c *C) {
 	}
 }
 
+func (s *partitionTableSuite) TestIdexMerge(c *C) {
+	if israce.RaceEnabled {
+		c.Skip("exhaustive types test, skip race test")
+	}
+
+	tk := testkit.NewTestKitWithInit(c, s.store)
+	tk.MustExec("create database test_idx_merge")
+	tk.MustExec("use test_idx_merge")
+	tk.MustExec("set @@tidb_partition_prune_mode = 'dynamic'")
+
+	// list partition table
+	tk.MustExec(`create table tlist(a int, b int, index idx_a(a), index idx_b(b)) partition by list(a)(
```

Comment:
Do we need to consider the clustered index?

Reply:
I already have added some cases about `clustered-index + dynamic-mode` in https://github.com/pingcap/tidb/pull/24523, should we consider `clusered-index + dynamic-mode + index-merge` particularly?

Reply:
I think so. Maybe we need to consider cross-feature.

Reply:
Ok. I'll update the ddl to use both clustered index and non clustered index.